File: sdk/javascript/src/api/DefaultApi.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient', '../model/CodeSnippet', '../model/CodeSnippetList'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('../model/CodeSnippet'), require('../model/CodeSnippetList'));
  } else {
    // Browser globals (root is window)
    if (!root.CodeSnipRestApi) {
      root.CodeSnipRestApi = {};
    }
    root.CodeSnipRestApi.DefaultApi = factory(root.CodeSnipRestApi.ApiClient, root.CodeSnipRestApi.CodeSnippet, root.CodeSnipRestApi.CodeSnippetList);
  }
}(this, function(ApiClient, CodeSnippet, CodeSnippetList) {
  'use strict';

  var DefaultApi = function DefaultApi(apiClient) {
    this.apiClient = apiClient || ApiClient.default;

    var self = this;
    
    
    /**
     * Creates a code snippet.
     * Creates a code snippet in the specified language.
     * @param {CodeSnippet} opts['codeSnippetBody'] Code snippet object.
     * @param {function} callback the callback function, accepting three arguments: error, data, response
     */
    self.snipPost = function(opts, callback) {
      opts = opts || {};
      var postBody = opts['codeSnippetBody'];
      

      var pathParams = {
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = null;

      return this.apiClient.callApi(
        '/snip', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );

    }
    
    /**
     * Retrieves the specified code snippet.
     * Retrieves the specified code snippet.
     * @param {String} codeSnippetUuid Code snippet unique identifier.
     * @param {function} callback the callback function, accepting three arguments: error, data, response
     *   data is of type: CodeSnippet
     */
    self.snipCodeSnippetUuidGet = function(codeSnippetUuid, callback) {
      var postBody = null;
      
      // verify the required parameter 'codeSnippetUuid' is set
      if (codeSnippetUuid == null) {
        throw "Missing the required parameter 'codeSnippetUuid' when calling snipCodeSnippetUuidGet";
      }
      

      var pathParams = {
        'code_snippet_uuid': codeSnippetUuid
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = CodeSnippet;

      return this.apiClient.callApi(
        '/snip/{code_snippet_uuid}', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );

    }
    
    /**
     * Updates a code snippet.
     * Updates a code snippet changes.
     * @param {String} codeSnippetUuid Code snippet unique identifier.
     * @param {CodeSnippet} opts['codeSnippetBody'] Code snippet object.
     * @param {function} callback the callback function, accepting three arguments: error, data, response
     */
    self.snipCodeSnippetUuidPut = function(codeSnippetUuid, opts, callback) {
      opts = opts || {};
      var postBody = opts['codeSnippetBody'];
      
      // verify the required parameter 'codeSnippetUuid' is set
      if (codeSnippetUuid == null) {
        throw "Missing the required parameter 'codeSnippetUuid' when calling snipCodeSnippetUuidPut";
      }
      

      var pathParams = {
        'code_snippet_uuid': codeSnippetUuid
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = null;

      return this.apiClient.callApi(
        '/snip/{code_snippet_uuid}', 'PUT',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );

    }
    
    /**
     * Deletes the specified code snippet.
     * Deletes the specified code snippet.
     * @param {String} codeSnippetUuid Code snippet unique identifier.
     * @param {function} callback the callback function, accepting three arguments: error, data, response
     */
    self.snipCodeSnippetUuidDelete = function(codeSnippetUuid, callback) {
      var postBody = null;
      
      // verify the required parameter 'codeSnippetUuid' is set
      if (codeSnippetUuid == null) {
        throw "Missing the required parameter 'codeSnippetUuid' when calling snipCodeSnippetUuidDelete";
      }
      

      var pathParams = {
        'code_snippet_uuid': codeSnippetUuid
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = null;

      return this.apiClient.callApi(
        '/snip/{code_snippet_uuid}', 'DELETE',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );

    }
    
    /**
     * Retrieves all code snippets.
     * 
     * @param {function} callback the callback function, accepting three arguments: error, data, response
     *   data is of type: CodeSnippetList
     */
    self.snipsGet = function(callback) {
      var postBody = null;
      

      var pathParams = {
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = CodeSnippetList;

      return this.apiClient.callApi(
        '/snips', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );

    }
    
    
  };

  return DefaultApi;
}));

File: sdk/java/src/main/java/io/swagger/client/api/DefaultApi.java
package io.swagger.client.api;

import com.sun.jersey.api.client.GenericType;

import io.swagger.client.ApiException;
import io.swagger.client.ApiClient;
import io.swagger.client.Configuration;
import io.swagger.client.Pair;

import io.swagger.client.model.CodeSnippet;
import io.swagger.client.model.CodeSnippetList;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2016-03-11T02:09:38.462Z")
public class DefaultApi {
  private ApiClient apiClient;

  public DefaultApi() {
    this(Configuration.getDefaultApiClient());
  }

  public DefaultApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  public ApiClient getApiClient() {
    return apiClient;
  }

  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  
  /**
   * Creates a code snippet.
   * Creates a code snippet in the specified language.
   * @param codeSnippetBody Code snippet object.
   * @throws ApiException if fails to make API call
   */
  public void snipPost(CodeSnippet codeSnippetBody) throws ApiException {
    Object localVarPostBody = codeSnippetBody;
    
    // create path and map variables
    String localVarPath = "/snip".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();

    

    

    

    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      "application/json"
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] {  };

    
    apiClient.invokeAPI(localVarPath, "POST", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, null);
    
  }
  
  /**
   * Retrieves the specified code snippet.
   * Retrieves the specified code snippet.
   * @param codeSnippetUuid Code snippet unique identifier.
   * @return CodeSnippet
   * @throws ApiException if fails to make API call
   */
  public CodeSnippet snipCodeSnippetUuidGet(String codeSnippetUuid) throws ApiException {
    Object localVarPostBody = null;
    
    // verify the required parameter 'codeSnippetUuid' is set
    if (codeSnippetUuid == null) {
      throw new ApiException(400, "Missing the required parameter 'codeSnippetUuid' when calling snipCodeSnippetUuidGet");
    }
    
    // create path and map variables
    String localVarPath = "/snip/{code_snippet_uuid}".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "code_snippet_uuid" + "\\}", apiClient.escapeString(codeSnippetUuid.toString()));

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();

    

    

    

    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      "application/json"
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] {  };

    
    GenericType<CodeSnippet> localVarReturnType = new GenericType<CodeSnippet>() {};
    return apiClient.invokeAPI(localVarPath, "GET", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, localVarReturnType);
    
  }
  
  /**
   * Updates a code snippet.
   * Updates a code snippet changes.
   * @param codeSnippetUuid Code snippet unique identifier.
   * @param codeSnippetBody Code snippet object.
   * @throws ApiException if fails to make API call
   */
  public void snipCodeSnippetUuidPut(String codeSnippetUuid, CodeSnippet codeSnippetBody) throws ApiException {
    Object localVarPostBody = codeSnippetBody;
    
    // verify the required parameter 'codeSnippetUuid' is set
    if (codeSnippetUuid == null) {
      throw new ApiException(400, "Missing the required parameter 'codeSnippetUuid' when calling snipCodeSnippetUuidPut");
    }
    
    // create path and map variables
    String localVarPath = "/snip/{code_snippet_uuid}".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "code_snippet_uuid" + "\\}", apiClient.escapeString(codeSnippetUuid.toString()));

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();

    

    

    

    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      "application/json"
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] {  };

    
    apiClient.invokeAPI(localVarPath, "PUT", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, null);
    
  }
  
  /**
   * Deletes the specified code snippet.
   * Deletes the specified code snippet.
   * @param codeSnippetUuid Code snippet unique identifier.
   * @throws ApiException if fails to make API call
   */
  public void snipCodeSnippetUuidDelete(String codeSnippetUuid) throws ApiException {
    Object localVarPostBody = null;
    
    // verify the required parameter 'codeSnippetUuid' is set
    if (codeSnippetUuid == null) {
      throw new ApiException(400, "Missing the required parameter 'codeSnippetUuid' when calling snipCodeSnippetUuidDelete");
    }
    
    // create path and map variables
    String localVarPath = "/snip/{code_snippet_uuid}".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "code_snippet_uuid" + "\\}", apiClient.escapeString(codeSnippetUuid.toString()));

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();

    

    

    

    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      "application/json"
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] {  };

    
    apiClient.invokeAPI(localVarPath, "DELETE", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, null);
    
  }
  
  /**
   * Retrieves all code snippets.
   * 
   * @return CodeSnippetList
   * @throws ApiException if fails to make API call
   */
  public CodeSnippetList snipsGet() throws ApiException {
    Object localVarPostBody = null;
    
    // create path and map variables
    String localVarPath = "/snips".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();

    

    

    

    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      "application/json"
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] {  };

    
    GenericType<CodeSnippetList> localVarReturnType = new GenericType<CodeSnippetList>() {};
    return apiClient.invokeAPI(localVarPath, "GET", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, localVarReturnType);
    
  }
  
}
